packages/core/src/verify.ts

```diff
@@ -87,6 +90,23 @@ function resolveChainConfig(
   return chainConfig;
 }
 
+function getImportSourceNames(
+  sourceName: string,
+  buildInfo: BuildInfo
+): string[] {
+  const contractSource = buildInfo.input.sources[sourceName].content;
+  const { imports } = analyze(contractSource);
+
+  const importSources = imports.map((i) =>
+    path.join(path.dirname(sourceName), i).replaceAll("\\", "/")
```

Comment:
@alcuadrado because this is our build info are we able to assume that there are no remappings that need to be considered?

Reply:
Does it work with global imports like `import "@openzepellin/Owner.sol";`?

Reply:
> @alcuadrado because this is our build info are we able to assume that there are no remappings that need to be considered?

The remappigns are already included in the buildInfo, and they are correct, because the build already succeeded.

Reply:
What we need to do with remappings though, is to remove the unused ones, as those may create a privacy issue.

They way to do that would be something like this:

* For each import in a file, if it's global, check if it matches any of the input remappings. If it does, flag that remapping as used.
* After processing all the files, only include the used remappings.

Reply:
This implements the logic of choosing which remapping affects an import, if any: https://github.com/NomicFoundation/hardhat/blob/a3ada90ad7362660c98372c36779409121af39ac/packages/hardhat-core/src/utils/remappings.ts#L5-L14

Reply:
Note that in that logic, the length of the "match" is used to pick the right remapping if more than one matches.